tests/test_scalars.py

```diff
@@ -278,89 +318,97 @@ def test_timedelta_mul_div() -> None:
+    check(assert_type(td * md_int64_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td * md_float_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    check(assert_type(md_int * td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(md_float * td, pd.Timedelta), pd.Timedelta)
+    # pyright is wrong here ndarray.__mul__(Timedelta) is NotImplemented
     check(
-        assert_type(md3 * td, np.ndarray),  # pyright: ignore[reportGeneralTypeIssues]
+        assert_type(
+            md_ndarray_intp * td, np.ndarray  # pyright: ignore[reportGeneralTypeIssues]
+        ),
         np.ndarray,
     )
     check(
-        assert_type(md4 * td, np.ndarray),  # pyright: ignore[reportGeneralTypeIssues]
+        assert_type(
+            md_ndarray_float * td,  # pyright: ignore[reportGeneralTypeIssues]
+            np.ndarray,
+        ),
         np.ndarray,
     )
-    check(assert_type(md5 * td, TimedeltaSeries), pd.Series)
-    check(assert_type(md6 * td, TimedeltaSeries), pd.Series)
-    check(assert_type(md7 * td, pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(assert_type(md8 * td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(mp_series_int * td, TimedeltaSeries), pd.Series)
+    check(assert_type(md_series_float * td, TimedeltaSeries), pd.Series)
+    check(assert_type(md_int64_index * td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(md_float_index * td, pd.TimedeltaIndex), pd.TimedeltaIndex)
 
     check(assert_type(td // td, int), int)
     check(assert_type(td // pd.NaT, float), float)
-    check(assert_type(td // md1, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td // md2, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td // md3, npt.NDArray[np.timedelta64]), np.ndarray)
-    check(assert_type(td // md4, npt.NDArray[np.timedelta64]), np.ndarray)
-    check(assert_type(td // md5, TimedeltaSeries), pd.Series)
-    check(assert_type(td // md6, TimedeltaSeries), pd.Series)
-    check(assert_type(td // md7, pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(assert_type(td // md8, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td // md_int, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td // md_float, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td // md_ndarray_intp, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td // md_ndarray_float, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td // mp_series_int, TimedeltaSeries), pd.Series)
+    check(assert_type(td // md_series_float, TimedeltaSeries), pd.Series)
+    check(assert_type(td // md_int64_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td // md_float_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td // md_timedelta_series, "pd.Series[int]"), pd.Series, int)
 
     check(assert_type(pd.NaT // td, float), float)
     # Note: None of the reverse floordiv work
     # TypeError: md1, md2, md3, md4, md5, md6, md7, md8
```

Comment:
update comment to refer to the new names

Reply:
Fixed.